contingency/contingency-api/src/main/java/com/powsybl/contingency/json/JsonContingencyListLoader.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.powsybl.contingency.json;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.auto.service.AutoService;
+import com.powsybl.commons.json.JsonUtil;
+import com.powsybl.contingency.ContingencyList;
+import com.powsybl.contingency.ContingencyListLoader;
+import com.powsybl.contingency.DefaultContingencyList;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * @author Mathieu Bague <mathieu.bague@rte-france.com>
+ */
+@AutoService(ContingencyListLoader.class)
+public class JsonContingencyListLoader implements ContingencyListLoader {
+
+    @Override
+    public String getFormat() {
+        return "json";
+    }
+
+    @Override
+    public ContingencyList load(String name, InputStream stream) throws IOException {
+        ObjectMapper objectMapper = JsonUtil.createObjectMapper();
+        objectMapper.registerModule(new ContingencyJsonModule());
+
+        return objectMapper.readValue(stream, DefaultContingencyList.class);
```

Comment:
Will be done in different PR linked with new issue '#1533' based on your comment

Reply:
@sylvlecl is right and it has to be done in this PR otherwise we won't be backward compatible when adding new contingency list type. The solution with @c is really ugly (I known this is what jackson do for that). We should add a type field in the list to store the contingency list type.

When we will add for example an automatic contingency list based on filtering criteria (like country, voltage level), serialization will be json too. How we will be able to deserialize it ? Same contingency loader right because associated to the json extension? So we need to add a type field? But in that case it is not really expandable as it will be hardcoded in the Json contingency list loader the mapping between the type and the contingency list subclass?